packages/flame/lib/src/text/sprite_font_renderer.dart

```diff
@@ -0,0 +1,144 @@
+/// to render text.
+///
+/// For example, suppose there is a spritesheet with sprites for characters from
+/// A to Z. Mapping these sprites into a [SpriteFontRenderer] allows then to
+/// write text using these sprite images as the font.
+///
+/// Currently, this class supports monospace fonts only -- the width and the
+/// height of all characters must be the same ([charWidth] and [charHeight]).
+/// Extra space between letters can be added via the [letterSpacing] parameter
+/// (it can also be negative to "squish" characters together). Finally, the
+/// [scale] parameter allows scaling the font to be bigger/smaller relative to
+/// its size in the source image.
+///
+/// The [paint] parameter is used to composite the character images onto the
+/// canvas. Its default value will draw the character images as-is. Changing
+/// the opacity of the paint's color will make the text semi-transparent.
+class SpriteFontRenderer extends TextRenderer {
+  SpriteFontRenderer(this._data, {this.letterSpacing = 0});
+
+  final SpriteFontData _data;
+  final double letterSpacing;
+
+  Image get source => _data.source;
+  double get charWidth => _data.charWidth;
+  double get charHeight => _data.charHeight;
+  double get scale => _data.scale;
+  bool get isMonospace => true;
+
+  Paint paint = Paint()..color = const Color(0xFFFFFFFF);
+
+  @override
+  double measureTextHeight(String text) => charHeight * scale;
+
+  @override
+  double measureTextWidth(String text) {
+    final n = text.length;
+    return n > 0 ? charWidth * scale * n + letterSpacing * (n - 1) : 0;
+  }
+
+  @override
+  Vector2 measureText(String text) {
+    return Vector2(measureTextWidth(text), measureTextHeight(text));
+  }
+
+  @override
+  void render(
+    Canvas canvas,
+    String text,
+    Vector2 position, {
+    Anchor anchor = Anchor.topLeft,
+  }) {
+    final rstTransforms = Float32List(4 * text.length);
+    final rects = Float32List(4 * text.length);
+    var j = 0;
+    var x0 = position.x;
+    final y0 = position.y;
+    for (final glyph in _textToGlyphs(text)) {
+      rects[j + 0] = glyph.srcLeft;
+      rects[j + 1] = glyph.srcTop;
+      rects[j + 2] = glyph.srcRight;
+      rects[j + 3] = glyph.srcBottom;
+      rstTransforms[j + 0] = glyph.rstSCos;
+      rstTransforms[j + 1] = glyph.rstSSin;
+      rstTransforms[j + 2] = x0 + glyph.rstTx;
+      rstTransforms[j + 3] = y0 + glyph.rstTy;
+      x0 += glyph.width + letterSpacing;
+      j += 4;
+    }
+    canvas.drawRawAtlas(source, rstTransforms, rects, null, null, null, paint);
+  }
+
+  Iterable<_GlyphInfo> _textToGlyphs(String text) {
+    return text.codeUnits.map(_getGlyphFromCodeUnit);
+  }
+
+  _GlyphInfo _getGlyphFromCodeUnit(int i) {
+    final glyph = _data._glyphs[i];
+    if (glyph == null) {
+      throw ArgumentError('No glyph for character "${String.fromCharCode(i)}"');
+    }
+    return glyph;
+  }
+}
+
+/// Helper class for creating [SpriteFontRenderer] instances.
+class SpriteFontData {
```

Comment:
I think this isn't quite the way I suggested, sorry if I wasn't clear on my previous comments, I will add a snippet of how I expect this API to look like:

```dart
final renderer = SpriteFontRenderer({
  SpriteFontData(
    source: image,
    charWidth: 6,
    charHeight: 8,
    glyphs: {
     'a': GlyphData(top: 0, left: 0),
     'b': GlyphData(top: 0, left: 6),
     ...
    },
});
```

That way this will be truly immutable and we don't need to imperatively call an `addGlyph` method.

Reply:
Alright, implemented this approach